sdk/python/kfp/v2/dsl/_pipeline.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2018-2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+from kfp.dsl import _container_op
+from kfp.dsl import _resource_op
+from kfp.dsl import _ops_group
+from kfp.v2.dsl.component_bridge import _create_container_op_from_component_and_arguments
+from kfp.components import _components
+from kfp.components._naming import _make_name_unique_by_adding_index
+import sys
+
+
+# This handler is called whenever the @pipeline decorator is applied.
+# It can be used by command-line DSL compiler to inject code that runs for every pipeline definition.
+_pipeline_decorator_handler = None
+
+
+def pipeline(name : str = None, description : str = None):
+  """Decorator of pipeline functions.
+
+  Example
+    ::
+
+      @pipeline(
+        name='my awesome pipeline',
+        description='Is it really awesome?'
+      )
+      def my_pipeline(a: PipelineParam, b: PipelineParam):
+        ...
+  """
+  def _pipeline(func):
+    if name:
+      func._component_human_name = name
+    if description:
+      func._component_description = description
+
+    if _pipeline_decorator_handler:
+      return _pipeline_decorator_handler(func) or func
+    else:
+      return func
+
+  return _pipeline
+
+class PipelineConf():
+  """PipelineConf contains pipeline level settings
+  """
+  def __init__(self):
```

Comment:
I think we can directly raise `NotImplementedError` here since for now nothing is actually supported.

Reply:
Gave a quick attempt, it throws here https://github.com/kubeflow/pipelines/blob/0795597562e076437a21745e524b5c960b1edb68/sdk/python/kfp/dsl/_pipeline.py#L225

I'll defer this to a later PR.

Reply:
SG. Do you mind dropping a TODO?

Reply:
Or actually since we're under v2 space maybe simply commenting out/removing all the attributes for now?